hera_pspec/pspecbeam.py

```diff
@@ -255,24 +257,38 @@ def get_Omegas(self, pols):
         OmegaPP : array_like
             Array containing power_sq_beam_int, shape: (Nbeam_freqs, Npols).
         """
-        # type check
-        if isinstance(pols, (int, np.int, np.int32)):
-            pols = [pols]
-        elif isinstance(pols, (str, np.str)):
-            pols = [pols]
-        if isinstance(pols, (list, np.ndarray, tuple)):
-            if isinstance(pols[0], (int, np.int, np.int32)):
-                pols = map(lambda p: uvutils.polnum2str(p), pols)
-                
-        # initialize blank lists
+        # Unpack polpairs into tuples
+        if not isinstance(polpairs, (list, np.ndarray)):
+            if isinstance(polpairs, (tuple, int, np.integer)):
+                polpairs = [polpairs,]
+            else:
+                raise TypeError("polpairs is not a list of integers or tuples")
+        
+        # Convert integers to tuples
+        polpairs = [uvputils.polpair_int2tuple(p) 
+                        if isinstance(p, (int, np.integer, np.int32)) else p 
+                        for p in polpairs]
+        
+        # Calculate Omegas for each pol pair
         OmegaP, OmegaPP = [], []
-        for p in pols:
-            OmegaP.append(self.power_beam_int(pol=p))
-            OmegaPP.append(self.power_beam_sq_int(pol=p))
+        for pol1, pol2 in polpairs:
+            if isinstance(pol1, (int, np.integer)):
+                pol1 = uvutils.polnum2str(pol1)
+            if isinstance(pol2, (int, np.integer)):
+                pol2 = uvutils.polnum2str(pol2)
+            
+            # Check for cross-pol; only auto-pol calculation currently supported
```

Comment:
looks like auto-pol is used here ;) hehe. perhaps same-pol?

Reply:
Done!